unblob/handlers/archive/ewf/ewf.py

```diff
@@ -0,0 +1,202 @@
+                    position
+                    + int.from_bytes(offset, byteorder="little")
+                    - DESCRIPTOR_LEN,
+                    sectors_per_chunk,
+                ):
+                    if compressed:
+                        compressed_chunk = zlib.decompress(chunk)
+                        output_file.write(compressed_chunk)
+                    output_file.write(chunk)
+
+    def extract(self, inpath: Path, outdir: Path):
+        with File.from_path(inpath) as file:
+            file.seek(EWF_HEADER_LEN)  # we skip the initial header
+            data_descriptor = self._struct_parser.parse(
+                "data_descriptor_t", file, Endian.LITTLE
+            )
+            logger.debug("data_descriptor_t", header=data_descriptor, _verbosity=3)
+
+            # the file is made of section, we loop over all the sections
+            while data_descriptor.definition != Definition.DONE.value:
+                file.seek(data_descriptor.next_offset, io.SEEK_SET)
+                data_descriptor = self._struct_parser.parse(
+                    "data_descriptor_t", file, Endian.LITTLE
+                )
+                logger.debug("data_descriptor_t", header=data_descriptor, _verbosity=3)
+
+                if data_descriptor.definition == Definition.VOLUME.value:
+                    volume_descriptor = self._struct_parser.parse(
+                        "volume_descriptor_t", file, Endian.LITTLE
+                    )
+                    sectors_per_chunk = find_chunk_size(volume_descriptor)
+
+                if data_descriptor.definition == Definition.SECTORS.value:
+                    position = file.tell()
+
+                if data_descriptor.definition == Definition.TABLE.value:
+                    self.table_descriptor(file, position, outdir, sectors_per_chunk)
+
+
+class EFWHandlerBase(StructHandler):
+    HEADER_STRUCT = "ewf_header_t"
+
+    def calculate_chunk(self, file: File, start_offset: int) -> ValidChunk:
+        header = self.parse_header(file, endian=Endian.LITTLE)
+
+        if not is_valid_header(header):
+            raise InvalidInputFormat("Invalid EWF header")
+
+        data_descriptor = self._struct_parser.parse(
+            "data_descriptor_t", file, Endian.LITTLE
+        )
+        while data_descriptor.definition != Definition.DONE.value:
+            file.seek(data_descriptor.next_offset, io.SEEK_SET)
+            data_descriptor = self._struct_parser.parse(
+                "data_descriptor_t", file, Endian.LITTLE
+            )
+
+        return ValidChunk(start_offset=start_offset, end_offset=file.tell())
+
+
+class EWFEHandler(EFWHandlerBase):
+    NAME = "ewfe"
+
+    PATTERNS = [HexString("45 56 46 09 0d 0a ff 00")]
+
+    C_DEFINITIONS = C_DEFINITIONS
+    HEADER_STRUCT = "ewf_header_t"
+    EXTRACTOR = EWFExtractor("ewf_header_t")
+
+
+class EWFLHandler(EFWHandlerBase):
+    NAME = "ewfl"
+
+    PATTERNS = [HexString("4C 56 46 09 0d 0a ff 00")]
```

Comment:
You can enforce a strict pattern and get rid of `is_valid_header`:

```suggestion
    PATTERNS = [HexString("4C 56 46 09 0d 0a ff 00 01 [2] 00")]
```